test/e2e/artifacts_test.go

```diff
@@ -224,6 +224,59 @@ func (s *ArtifactsSuite) TestStoppedWorkflow() {
 	}
 }
 
+func (s *ArtifactsSuite) TestFailedWorkflow() {
```

Comment:
The failed workflow is part of `fixtures.ToBeCompleted`
maybe we can include this in the https://github.com/argoproj/argo-workflows/blob/34967127bfe304da0f72dc1dba63558a1609b2d0/test/e2e/artifacts_test.go#L247 instead?